engine/src/main/java/io/zeebe/engine/processing/bpmn/behavior/BpmnIncidentBehavior.java

```diff
@@ -67,41 +51,12 @@ private void createIncident(
         .setElementInstanceKey(context.getElementInstanceKey())
         .setElementId(context.getElementId())
         .setVariableScopeKey(variableScopeKey)
-        .setErrorType(errorType)
-        .setErrorMessage(errorMessage);
-
-    // todo: remove storing record after incident is migrated
-    if (!MigratedStreamProcessors.isMigrated(context.getBpmnElementType())) {
-      final var intent = determineCommandIntent(context);
-      elementInstanceState.storeRecord(
-          context.getElementInstanceKey(),
-          context.getFlowScopeKey(),
-          context.getRecordValue(),
-          intent,
-          Purpose.FAILED);
-    }
+        .setErrorType(failure.getErrorType())
+        .setErrorMessage(failure.getMessage());
 
     streamWriter.appendNewCommand(IncidentIntent.CREATE, incidentCommand);
   }
 
-  private ProcessInstanceIntent determineCommandIntent(final BpmnElementContext context) {
-    if (!MigratedStreamProcessors.isMigrated(context.getBpmnElementType())) {
-      return context.getIntent();
-    } else {
-      if (context.getIntent() == ProcessInstanceIntent.ELEMENT_ACTIVATING) {
-        return ProcessInstanceIntent.ACTIVATE_ELEMENT;
-      } else if (context.getIntent() == ProcessInstanceIntent.ELEMENT_COMPLETING) {
-        return ProcessInstanceIntent.COMPLETE_ELEMENT;
-      }
-    }
-    throw new UnsupportedOperationException(
-        String.format(
-            "Expected to raise incident, but element state %s not one of incident supporting states %s %s",
-            context.getIntent(),
-            ProcessInstanceIntent.ELEMENT_ACTIVATING,
-            ProcessInstanceIntent.ELEMENT_COMPLETING));
-  }
-
   public void resolveIncidents(final BpmnElementContext context) {
     incidentState.forExistingProcessIncident(
         context.getElementInstanceKey(),
```

Comment:
I have seen during the sub process migration that it is used on terminating. Does it make sense here to use the state writer?

Reply:
Do you want to do it here ? Or should I do it on my migration?

Reply:
I'll have a look at this

Reply:
I don't think that's should be part of this PR. Let's do it on the first element processor that needs that.

Reply:
ok so sub processor :laughing: